OpenHardwareMonitor/GUI/StartupManager.cs

```diff
@@ -217,31 +221,40 @@ private bool CreateSchedulerTask(bool atBootup)
 
         private void DeleteSchedulerTask()
         {
+            int taskCount = -1;
+            
             ITaskFolder root = scheduler.GetFolder("\\");
             try
             {
                 ITaskFolder folder = root.GetFolder("Open Hardware Monitor");
                 folder.DeleteTask("Startup", 0);
+                taskCount = folder.GetTasks(0x1).Count; //0x1 = TASK_ENUM_HIDDEN
             }
-            catch (IOException)
+            catch (IOException ex)
             {
+                _logger.LogError(ex, ex.Message);
             }
 
             try
             {
-                root.DeleteFolder("Open Hardware Monitor", 0);
+                // Previous versions might have registered under this name
+                root.DeleteTask("OpenHardwareMonitorAutoStart", 0); 
             }
-            catch (IOException)
+            catch (IOException ex)
             {
+                _logger.LogError(ex, ex.Message);
             }
 
-            try
-            {
-                // Previous versions might have registered under this name
-                root.DeleteTask("OpenHardwareMonitorAutoStart", 0);
-            }
-            catch (IOException)
-            {
+            if (0 < taskCount)
```

Comment:
Shouldn't this be `if (0 == taskCount)`? This deletes the folder whenever taskCount is > 0

Reply:
> Shouldn't this be `if (0 == taskCount)`? This deletes the folder whenever taskCount is > 0

Yes, you are right. It should be 0==taskCount.
I wonder if checking for task count even makes sense, other than reducing the amount of exceptions raised, since we try to delete folder in a try\catch block anyway. Maybe its pointless overcomplication.

Task updating block may not occur if the folder does not contain "Startup" task, leaving it to "-1". If the folder is also empty, since the condition wouldnt be met, the folder won't get deleted.

I'm unaware of try\catch overhead in C#, but I'd suggest two options:
1. The only thing we change, compared to the release version, is add one other type to exception handling+add logging. So we dont check task count, since we do everything in try\catch blocks anyway.
2. We add local bool variable, check if patch exists in try\catch block, update the bool variable if it is, and do braching based on said variable: trying to delete "Startup" task and\or folder only may occur if the folder itself exists. It may make code harder to read. It also may speed up the code, if try\catch blocks and/or Win32 API overhead is a thing.

Reply:
Try/catch blocks have some overhead (or rather, handling the exception has), but since this code runs only once and not in a loop, that is negligible. You can go with solution 1.

Reply:
I did as you suggested.